exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/agg/TestHashAggEmitOutcome.java

```diff
@@ -0,0 +1,789 @@
+import org.apache.drill.exec.record.RecordBatch;
+import org.apache.drill.exec.record.VectorContainer;
+import org.apache.drill.exec.record.VectorWrapper;
+import org.apache.drill.exec.record.metadata.TupleMetadata;
+import org.apache.drill.test.rowSet.RowSet;
+import org.apache.drill.test.rowSet.schema.SchemaBuilder;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.drill.exec.record.RecordBatch.IterOutcome.EMIT;
+import static org.apache.drill.exec.record.RecordBatch.IterOutcome.OK_NEW_SCHEMA;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Category(OperatorTest.class)
+public class TestHashAggEmitOutcome extends BaseTestOpBatchEmitOutcome {
+  //private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(TestHashAggEmitOutcome.class);
+
+  /**
+   * Verifies count of column in the received batch is same as expected count of columns.
+   * @param batch - Incoming record batch
+   * @param expectedColCount - Expected count of columns in the record batch
+   */
+  private void verifyColumnCount(RecordBatch batch, int expectedColCount) {
+    assertEquals(String.format("Actual number of columns: %d is different than expected count: %d",
+      batch.getContainer().getNumberOfColumns() ,expectedColCount),
+      batch.getContainer().getNumberOfColumns() ,expectedColCount);
+  }
+
+  /**
+   * Verifies the data received in incoming record batch with the expected data stored inside the expected batch.
+   * Assumes the expected batch values are in the same order as the incoming batch.
+   * @param inBatch - incoming record batch
+   * @param expectedBatch - expected record batch with expected data
+   */
+  private void verifyBaseline(RecordBatch inBatch, VectorContainer expectedBatch) {
+    int numCols = inBatch.getContainer().getNumberOfColumns();
+
+    for (int ind = 0; ind < inBatch.getRecordCount(); ind++) {
+      List<String> expValue = new ArrayList<>(numCols);
+      List<String> value = new ArrayList<>(numCols);
+
+      for (VectorWrapper<?> vw : inBatch) {
+        Object o = vw.getValueVector().getAccessor().getObject(ind);
+        decodeAndAddValue(o, value);
+      }
+
+      for (VectorWrapper<?> vw : expectedBatch) {
+        Object e = vw.getValueVector().getAccessor().getObject(ind);
+        decodeAndAddValue(e, expValue);
+      }
+      assertTrue(String.format("Expected value in position %d did not matches the actual value",ind),
+        expValue.equals(value));
+    }
+  }
+
+  private void decodeAndAddValue(Object currentValue, List<String> listToAdd) {
+    if (currentValue == null) {
+      listToAdd.add("null");
+    } else if (currentValue instanceof byte[]) {
+      listToAdd.add(new String((byte[]) currentValue));
+    } else {
+      listToAdd.add(currentValue.toString());
+    }
+  }
```

Comment:
Actually I just found out that all the above functions is not needed instead we can use below methods for verification. I am also updating TopN tests to use it.
```
RowSet actualRowSet = DirectRowSet.fromContainer(topNBatch.getContainer());
new RowSetComparison(expectedRowSet).verify(actualRowSet);
```